vulcan/lib/server/models/workflow.rb

```diff
@@ -0,0 +1,150 @@
+# A vulcan session model.  For now this is just static, but could be made backed by a database, or just about anything we like.
+class Workflow
+  attr_reader :workflow_name, :raw, :description, :display_name
+
+  def initialize(workflow_name, raw, description: "", display_name: "")
+    @workflow_name = workflow_name
+    @raw = raw
+    @description = description
+    @display_name = display_name
+  end
+
+  def self.from_yaml_file(filename, description: "", display_name: "")
+    Workflow.new(
+        ::File.basename(filename),
+        YAML.safe_load(
+            File.join(
+                File.dirname(__FILE__),
+                "../workflows/#{filename}"
+            )
+        ),
+        description: description,
+        display_name: display_name,
+    )
+  end
+
+  # Returns a list of lists, describing the dependency between steps, primary inputs, and primary outputs.
+  # Each inner list is a unique path in side of the workflow starting a primary_inputs and terminating
+  # at either a primary_output or a step that is not used as an input the workflow
+  def unique_paths
```

Comment:
Will the step name be included in the paths? It's the key in the YAML, but we also need it in the UI to display to the user, so don't want to lose it in the YAML conversion...

Reply:
Similar question for the outputs ... especially if there are multiple outputs at a given step, I think we'll need to include the output file name in each object in the `out` Array, so the user knows which output they are looking at.

Reply:
Don't worry, this isn't yet the part that connects to the api, this is more a functional part that helps me back the api.

I will keep all the information you need preserved when I return the actual json.

Reply:
Yes, I will preserve the filenames in all the outputs, I assure you.  This model is backing to other functionality, i haven't gotten to the part where this ties directly into the api yet.